### PR TITLE
fix: countryCode property in Input type phone number

### LINKED ISSUE #7475: [Bug] Input type phone number, Country code is not accessible
## Description

The newly introduced phone number type in the input widget does not allow the end user to access the the selected country code. The API is exposed but does not return any value.

### Important Details

- Version: [Cloud / Self-Hosted vx.x]
- OS: [e.g.MacOSX]
- Environment [production

### PR DESCRIPTION
**Pull request by vicky-primathon**
## Description
Fix countryCode property in Input type phone number

Fixes #7475 

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: fix/selected-phone-number-country-code 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 54.81 **(0.03)** | 36.68 **(0.07)** | 33.54 **(0.02)** | 55.34 **(0.02)**
 :green_circle: | app/client/src/utils/DSLMigrations.ts | 73.78 **(0.16)** | 65.09 **(-0.05)** | 70.21 **(0)** | 73.59 **(0.16)**
 :green_circle: | app/client/src/utils/autocomplete/TernServer.ts | 52.26 **(0.22)** | 41.25 **(0.83)** | 34.48 **(0)** | 56.22 **(0.26)**
 :sparkles: :new: | **app/client/src/utils/migrations/InputWidget.ts** | **80** | **80** | **75** | **79.17**
 :green_circle: | app/client/src/widgets/InputWidget/widget/index.tsx | 64.52 **(2.76)** | 43.75 **(2.08)** | 68.18 **(0)** | 67.05 **(3.13)**</details>

### CODE REVIEW

**Comment by riodeuno** [comment]
@somangshu I don't think this completely solve the issue here. When I try to bind the `Input1.text` I don't get the country code. There is also no other property which has the country code. Not sure, if this was a part of the original issue. Please review the deploy preview.

**Comment by somangshu** [comment]
@riodeuno I can see that for the type phone number the `{{input.countryCode}}` works and returns the short code, This however should return the int code and not the short code. 

@vicky-primathon I see that this does not function well in case of the currency as well. We need to fix the issue.

**Review by somangshu — CHANGES_REQUESTED** [review]
Change as mentioned above

**Comment by vicky-primathon** [comment]
> @riodeuno I can see that for the type phone number the `{{input.countryCode}}` works and returns the short code, This however should return the int code and not the short code.
> 
> @vicky-primathon I see that this does not function well in case of the currency as well. We need to fix the issue.

@somangshu  in case of currency, selected country code can be accessed as {{input.currencyCountryCode}} 
Should we expose single api i.e {{input.countryCode}} for both phone number and currency selected country code.

**Comment by somangshu** [comment]
> @somangshu in case of currency, selected country code can be accessed as {{input.currencyCountryCode}}
> Should we expose single api i.e {{input.countryCode}} for both phone number and currency selected country code.

@vicky-primathon when I check:
- `currencyCountryCode` does not return anything. It should return the ISO code  Screenshot 2021-09-23 at 4 02 14 PM
- `currencyCountryCode` should be changed to `currencyCode`.
- The `countryCode` should return the phone extension ex: +91

Let me know if you have any doubts

**Review by somangshu — DISMISSED** [review]
Deploy preview looks good to me, All changes as requested are updated.

@riodeuno can you please verify the code changes

File: app/client/src/widgets/InputWidget/component/ISDCodeDropdown.tsx [hunk 1]
```diff
@@ -36,8 +36,8 @@ const getISDCodeOptions = (): Array<DropdownOption> => {
       leftElement: countryToFlag(item.code),
       searchText: item.name,
       label: `${item.name} (${item.dial_code})`,
-      value: item.code,
-      id: item.dial_code,
+      value: item.dial_code,
```
Comment (riodeuno) [inline app/client/src/widgets/InputWidget/component/ISDCodeDropdown.tsx:39]: @vicky-primathon Will these changes effect existing users? How do we verify this in the UI?

Comment (vicky-primathon) [reply app/client/src/widgets/InputWidget/component/ISDCodeDropdown.tsx:39]: Will have to write migrations to update default selected code value from country code to dial_code

Comment (vicky-primathon) [reply app/client/src/widgets/InputWidget/component/ISDCodeDropdown.tsx:39]: @riodeuno added migrations for this

Comment (riodeuno) [reply app/client/src/widgets/InputWidget/component/ISDCodeDropdown.tsx:39]: @vivekverma2312 We will need to verify the migrations during QA. The approach would be open a bound input widget in this deploy preview, which was created in release.

File: app/client/src/utils/migrations/InputWidget.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  currentDSL: DSLWidget,
+) => {
+  currentDSL.children = currentDSL.children?.map((child: WidgetProps) => {
+    if (child.type === "INPUT_WIDGET") {
+      if (child.inputType === "PHONE_NUMBER" && child.phoneNumberCountryCode) {
+        const ISDCodeOption = ISDCodeOptions.find((item: ISDCodeProps) => {
+          return item.code === child.phoneNumberCountryCode;
+        });
+        // eslint-disable-next-line no-console
+        console.log({ ISDCodeOption });
```
Comment (riodeuno) [inline app/client/src/utils/migrations/InputWidget.ts:15]: We have a console log left over here.

**Review by riodeuno — CHANGES_REQUESTED** [review]

**Review by riodeuno — APPROVED** [review]

**Comment by riodeuno** [comment]
Note for QA: We need a regression test for these types of input widgets.
cc @YogeshJayaseelan @vivekverma2312

**Comment by vivekverma2312** [comment]
@vicky-primathon 
1. If input field is empty then `{{Input1.countryCode}}  - {{Input1.text}}` is showing undefined in text box.

Screenshot 2021-10-11 at 3 48 20 PM

2. During migration `{{Input2.currencyCountryCode}}{{Input2.text}}` is not migrating to `{{Input2.currencyCode}} {{Input2.text}}`. Due to this changing currency in Input widget do not change the currency in text widget.

**Comment by somangshu** [comment]
@vicky-primathon @riodeuno I disagree to CountryCode changing to DialCode, this is not the generic convention. 

cc @Nikhil-Nandagopal 

Also since this is blocked by the backend migration, should we close this or draft it until unblocked

**Comment by riodeuno** [comment]
@somangshu I don't have strong opinions on this. `countryCode` works. 
@vicky-primathon Let's make this into a draft for now.

**Comment by somangshu** [comment]
@riodeuno I said that because the name used in the first release used `countryCode` and if we continue with it we will not need any migrations.

**Comment by sbalaji1192** [comment]
@vicky-primathon Lets close this PR since we're gonna introduce a v2 of input widget. We will do these changes on the v2.

**Pull request closed without merging**